crates/starknet/src/transactions.rs

```diff
@@ -157,15 +158,26 @@ impl StarknetTransaction {
     }
 
     pub fn get_receipt(&self) -> DevnetResult<TransactionReceipt> {
-        let transaction_events = self.get_events();
+        let events = self.get_events();
+
+        let mut messages_sent: Vec<MessageToL1> = vec![];
+        for message in self.get_l2_to_l1_messages().into_iter() {
+            messages_sent.push(message.try_into()?);
+        }
+
+        let output = TransactionOutput {
+            actual_fee: self.execution_info.actual_fee,
+            messages_sent,
+            // TODO: we may have performances issues here? Can have a lot's of events..
```

Comment:
I would keep the comment but I would remove TODO. If in the future there will be a performance issue related to l1l2, we will take a look here anyway.

Reply:
@glijm feel free to remove the `clone` method invocation